src/rest/rest_web_server.cpp

```diff
@@ -0,0 +1,233 @@
+/*
+ *  Copyright (c) 2020, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "rest/rest_web_server.hpp"
+
+#include <algorithm>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <sys/time.h>
+
+using std::chrono::duration_cast;
+using std::chrono::microseconds;
+using std::chrono::steady_clock;
+
+namespace otbr {
+namespace rest {
+
+const unsigned RestWebServer::kMaxServeNum             = 100;
+const unsigned RestWebServer::kTimeout                 = 1000000;
+const unsigned RestWebServer::kScaleForCallbackTimeout = 4;
+const unsigned RestWebServer::kPortNumber              = 80;
+
+RestWebServer::RestWebServer(ControllerOpenThread *aNcp)
+    : mNcp(aNcp)
+    , mAddress(new sockaddr_in())
+
+{
+    mResource = std::unique_ptr<Resource>(new Resource(aNcp, kTimeout * kScaleForCallbackTimeout));
+}
+
+otbrError RestWebServer::Init()
+{
+    int       ret;
+    otbrError error = OTBR_ERROR_NONE;
+    mResource->Init();
+    mAddress->sin_family      = AF_INET;
+    mAddress->sin_addr.s_addr = INADDR_ANY;
+    mAddress->sin_port        = htons(kPortNumber);
+    mListenFd                 = socket(AF_INET, SOCK_STREAM, 0) == -1;
```

Comment:
```suggestion
    mListenFd                 = socket(AF_INET, SOCK_STREAM, 0);
```
Can't imagine how this works.

Reply:
Oh